src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -15,15 +22,110 @@ public FileContentResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileContentResult result)
         {
+            RangeItemHeaderValue range = new RangeItemHeaderValue(0, 0);
+            long rangeLength = default(long);
+            if (result.EnableRangeProcessing)
+            {
+                range = SetContentRangeAndStatusCode(context, result);
+                rangeLength = (range == null) ? 0 : SetRangeHeaders(context, result, range);
+            }
+
             SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            return WriteFileAsync(context, result, range, rangeLength);
+        }
+
+        private static Task WriteFileAsync(ActionContext context, FileContentResult result, RangeItemHeaderValue range, long rangeLength)
+        {
+            var response = context.HttpContext.Response;
+            var outputStream = response.Body;
+
+            if (!result.EnableRangeProcessing)
+            {
+                return response.Body.WriteAsync(result.FileContents, offset: 0, count: result.FileContents.Length);
+            }
+
+            else if (range == null || rangeLength == 0)
+            {
+                return Task.CompletedTask;
+            }
+
+            else
+            {
+                try
+                {
+                    return response.Body.WriteAsync(result.FileContents, offset: (int)range.From.Value, count: (int)rangeLength);
+                }
+
+                catch (OperationCanceledException ex)
+                {
+                    // Don't throw this exception, it's most likely caused by the client disconnecting.
+                    // However, if it was cancelled for any other reason we need to prevent empty responses.
+                    context.HttpContext.Abort();
+                    return Task.FromException(ex);
+                }
+
+            }
         }
 
-        private static Task WriteFileAsync(ActionContext context, FileContentResult result)
+        private RangeItemHeaderValue SetContentRangeAndStatusCode(ActionContext context, FileContentResult result)
```

Comment:
Seems like this method is duped and not trivial, can you figure out a way to share this logic?